### PR TITLE
seq2seq and InternalTimeDistributed bug fix

### PR DESCRIPTION
**Pull request by dding3**
This PR includes several seq2seq bug fix which included in https://github.com/intel-analytics/analytics-zoo/pull/1062/files. Just in case PR #1062 cannot be merged before 0.4 release.

1. RNNDecoder should not update input
2. seq2seq infer should take into account `generator` is null
3. InternalTimeDistributed should compute `outputsize` in each `forward`

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/InternalTimeDistributed.scala [hunk 1]
```diff
@@ -96,13 +96,12 @@ private[zoo] class InternalTimeDistributed[T: ClassTag](
...
     val steps = _inputSize(1)
 
-    if (outputSize == null) {
-      val combinedShape = _output.size()
-      require(combinedShape(0) == batch * steps,
-        s"combined batch: ${combinedShape(0)} should match ${batch} * ${steps}")
-      val splitedShape = Array(batch, steps) ++ combinedShape.drop(1)
-      outputSize = splitedShape
-    }
+    val combinedShape = _output.size()
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/InternalTimeDistributed.scala]: The whole changes is only for removing `if (outputSize == null)`? any reason?

Comment (dding3) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/layers/InternalTimeDistributed.scala]: Yes, there is a chance in a batch time sequence is different, and we need compute outputSize in each forward.

**Comment by dding3** [comment]
Thanks for @glorysdj for running jekins. It passed. http://10.239.47.211:18888/job/ZOO-PR-Validation/1716/console

**Comment by dding3** [comment]
jekins passed. http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/1738/

**Pull request merged**